src/services/eventsService.ts

```diff
@@ -0,0 +1,84 @@
+import axios from 'axios'
+import { format, formatISO, isToday, parseISO, sub } from 'date-fns'
+import fi from 'date-fns/locale/fi'
+import * as R from 'remeda'
+
+import { addNewEvent, fetchPostedEvents } from '../db/eventDb'
+import { EventObject } from '../types'
+
+export const todaysEvents = async (): Promise<string> => {
+  const events = await retrieveEvents()
+  const eventsToday = R.filter(events, e => isToday(parseISO(e.starts)))
+
+  const registrationToday = R.filter(events, e =>
+    isToday(parseISO(e.registration_starts))
+  )
+
+  if (eventsToday?.length > 0 || registrationToday?.length > 0) {
+    return `*Tänään:* \n ${listEvents(eventsToday, 'HH:mm', false)} ${listEvents(
+      registrationToday,
+      'HH:mm',
+      true
+    )}`
+  }
+}
+
+export const pollEvents = async (): Promise<string> => {
+  const events = await retrieveEvents()
+  const filteredEvents = await filterPostedEvents(events)
+  const addedEvents = await R.pipe(filteredEvents, R.map(addNewEvent), promises =>
+    Promise.all(promises)
+  )
+  return newEvents(addedEvents)
+}
+
+const retrieveEvents = async () => {
+  const { data } = await axios.get<EventObject[]>(
+    `https://event-api.tko-aly.fi/api/events?fromDate=${formatISO(
+      sub(Date.now(), { months: 3 })
+    ).slice(0, -6)}` // Remove the timezone from the ISO date string
+  )
+
+  return R.filter(data, e => e.deleted === 0)
+}
+
+const newEvents = (events: EventObject[]) => {
+  if (events?.length === 0) return
```

Comment:
Hmm, wonder if optional chaining works here since `null === 0` return false.

Reply:
yeah, that was definitely broken, good catch!